src/Umbraco.Web.Common/AspNetCore/AspNetCoreHostingEnvironment.cs

```diff
@@ -10,12 +10,12 @@ namespace Umbraco.Web.Common.AspNetCore
 {
     public class AspNetCoreHostingEnvironment : Core.Hosting.IHostingEnvironment
     {
-        private IOptionsMonitor<HostingSettings>  _hostingSettings;
+        private IOptions<HostingSettings>  _hostingSettings;
         private readonly IWebHostEnvironment _webHostEnvironment;
 
         private string _localTempPath;
 
-        public AspNetCoreHostingEnvironment(IOptionsMonitor<HostingSettings> hostingSettings, IWebHostEnvironment webHostEnvironment)
+        public AspNetCoreHostingEnvironment(IOptions<HostingSettings> hostingSettings, IWebHostEnvironment webHostEnvironment)
```

Comment:
This change avoids us from updating the debug flag without restarting the app.

Reply:
Ergh, in which case I think really CoreRuntimeBootstrapper class needs to be nuked.

Everything it does should happen after ServiceProvider is built instead of this two pass approach to booting.

Will reset to fe3b57b (for which CI build has already completed successfully) and have another crack another day.